src/kiara_plugin/topic_modelling/modules/corpus_preparation.py

```diff
@@ -30,83 +25,221 @@ def create_inputs_schema(self):
             },
             "file_name_col": {
                 "type": "string",
-                "doc": "The column containing file names with metadata. In order to work, file names need to comply with LCCN pattern '/sn86069873/1900-01-05/' containing publication reference and date."
+                "doc": "The column containing file names with metadata. In order to work, file names need to comply with LCCN pattern '/sn86069873/1900-01-05/' containing publication reference and date.",
             },
             "map": {
                 "type": "list",
                 "doc": "List of lists of unique publications references and publication names in the collection provided in the same order.",
                 "optional": True,
-            }
+            },
         }
 
     def create_outputs_schema(self):
         return {
             "corpus_table": {
                 "type": "table",
-                "doc": "Augmented table containing extracted metadata."
+                "doc": "Augmented table containing extracted metadata.",
             }
         }
 
     def process(self, inputs, outputs) -> None:
-        
+        import re
+        import polars as pl
+        import pyarrow as pa
+
         table_obj = inputs.get_value_obj("corpus_table")
         column_name = inputs.get_value_obj("file_name_col").data
-        pub_refs, pub_names = None, None
 
-        try:
-            pub_refs = inputs.get_value_obj("map").data[0]
-            pub_names = inputs.get_value_obj("map").data[1]
+        if not column_name:
+            raise KiaraProcessingException("No file name column name provided.")
+        
+        if not table_obj:
+            raise KiaraProcessingException("No table provided.")
+        
+        if table_obj.is_set:
 
-        except Exception as e:
-            pass;
+            sources: Union[pa.Table, None] = table_obj.data
+            
+            assert sources is not None
 
+            sources_col_names = sources.column_names
+
+            if column_name not in sources_col_names:
+
+                raise KiaraProcessingException(
+                    f"Could not find file names column '{column_name}' in the table. Please specify a valid column name manually, using one of: {', '.join(sources_col_names)}"
+                )
+            
+            sources_data: pa.Table = table_obj.data.arrow_table
+            
+            sources_tb: pl.DataFrame = pl.from_arrow(sources_data) # type: ignore
```

Comment:
Again, probably better to name it `source_polars_format` or so, but that's a nitpick and you can just ignore me. More importantly, the [`KiaraTable`](https://github.com/DHARPA-Project/kiara_plugin.tabular/blob/d7117fe61b2c519fb573d18b71133ece3d2e4a5e/src/kiara_plugin/tabular/models/table.py#L30) class I mentioned earlier does have a convenience method `to_polars_dataframe`, so you can just use that:

```
sources_polars_format = sources.to_polars_dataframe()
```
Your IDE should give you all the available methods because of the type hinting, I think. So often its worth having a look what it tells you is there, and browse through it (not just with kiara types, I do that all the time).